legion/engine/core/defaults/hierarchysystem.cpp

```diff
@@ -0,0 +1,44 @@
+#include <core/defaults/hierarchysystem.hpp>
+
+void legion::core::HierarchySystem::onPositionModified(events::component_modification<position>* event)
+{
+    /*if (event->entity == m_ignore)
+        return;*/
+
+    position diff = event->newValue - event->oldValue;
+    for (auto child : event->entity.children())
+        child.write_component<position>(child.read_component<position>() + diff);
+}
+
+void legion::core::HierarchySystem::onRotationModified(events::component_modification<rotation>* event)
+{
+    rotation diff = event->newValue * math::inverse(event->oldValue);
+    position p = event->entity.read_component<position>();
+    for (auto child : event->entity.children())
+    {
+        //m_ignore = child;
+        child.write_component<position>(p + (math::toMat3(diff) * (child.read_component<position>() - p)));
+        //m_ignore = invalid_id;
```

Comment:
These comments can probably go. More comments up ahead in this file. I will not commend on these